overlord/exportstate/special.go

```diff
@@ -0,0 +1,188 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package exportstate
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+)
+
+// special.go holds special cases when the export manifest is not stored in snap.yaml
+
+// snapdTools returns an export set describing snapd tools.
+//
+// basePath is the path of the "$libexecdir/snapd" directory. This directory
+// differs depending on who is providing the tools. Host with variable
+// $libexecdir or snaps with a fixed one.
+func snapdTools(basePath string) ExportSet {
```

Comment:
In case of snapd refresh, is this executed for the new snapd, so that any new tools that are unknown to the old snapd will be exported? I think the answer is yes, but double-checking.

Reply:
This will be re-executed by the new snapd as a part of the export manager startup process.